### PR TITLE
feat(fhir): add Practitioner and MedicationRequest R4 generators (#388)

### LINKED ISSUE #388: feat(fhir): add Practitioner and MedicationRequest FHIR R4 resource generators
## Context
Epic uses MedicationRequest (not MedicationStatement) for active prescriptions. CareBridge needs both to round-trip data with Epic. Additionally, Practitioner resources are needed to properly reference clinicians in exported bundles.

### Practitioner Generator (`generators/practitioner.ts`)
- Map `users` table (role=physician/specialist/nurse) → FHIR Practitioner
- Include name (parse into family/given), identifier (use user.id as internal identifier)
- Map `specialty` → qualification/specialty coding (SNOMED CT or NUCC)
- Map `department` → optional extension or PractitionerRole
- Only generate for clinical roles, not patients or admin

### MedicationRequest Generator (`generators/medication-request.ts`)
- Map CareBridge `medications` table → FHIR MedicationRequest
- Fields: status, intent (always "order"), medicationCodeableConcept (RxNorm from existing medication-statement.ts patterns), subject, authoredOn (started_at), requester (ordering_provider_id → Practitioner reference)
- Dosage instruction: route (SNOMED), frequency (text for now), doseAndRate
- This is the Epic-compatible format; MedicationStatement remains for recorded/historical meds

### Integration
- Export from `generators/index.ts`
- Practitioner references used in Encounter.participant, Procedure.performer, MedicationRequest.requester
- Include practitioners and medication requests in FHIR bundle export

## Acceptance criteria
- `toFhirPractitioner()` and `toFhirMedicationRequest()` exported
- Practitioner resources included in bundle (deduplicated — one per unique provider)
- Unit tests covering mapping logic
- TypeScript strict, ESM imports

## Dependencies
- #386 (FHIR R4 type definitions)

### PR DESCRIPTION
**Pull request by blamechris**
## Summary

- `services/fhir-gateway/src/generators/practitioner.ts` — only clinical roles (physician / specialist / nurse) produce a Practitioner. Name parsed into family + given via last-token-is-family heuristic; credentials (\", MD\" / \", RN\") stripped from the split but preserved in `text`. Specialty → text-only qualification coding (no NUCC code until schema stores one)
- `services/fhir-gateway/src/generators/medication-request.ts` — intent always \"order\" (CareBridge meds are concrete prescriptions). Status narrowed to the MedicationRequest value set. requester prefers `ordering_provider_id`, falls back to `prescribed_by`
- `exportPatient`: emits MedicationRequest alongside MedicationStatement (distinct `fullUrl` prefix `request-<id>`) so Epic-style consumers see orders and recorded-history consumers see statements. Collects all unique provider ids from Encounter / Procedure / MedicationRequest references, fetches matching `users` rows, emits one deduplicated Practitioner each. Filters out non-clinical roles
- 14 new tests (7 practitioner + 11 medication-request) covering status mappings, name edge cases (credentials, middle names, single-token), RxNorm fallback, requester priority

## Test plan

- [x] `pnpm --filter @carebridge/fhir-gateway test` — 158/158
- [x] `pnpm typecheck` — 40/40
- [x] `pnpm lint` — 22/22, no new warnings
- [x] Existing export-audit / headers / purge-warning tests still green (already handle the new parallel fetches via selectMock fallthrough)

Closes #388

**Depends on #922** via stacked branch order.

### CODE REVIEW

**Comment by blamechris** [comment]
## Review: PR #923 — feat(fhir): Practitioner + MedicationRequest R4 generators

**Verdict: Approve with follow-ups** — ship it; tracked concerns are non-blocking.

### What looks good
- Generator shape matches the existing pattern (Condition, MedicationStatement, AllergyIntolerance). Pure functions, ESM imports with `.js`, strict TS.
- `intent: "order"` is the right call for CareBridge's prescription model (no planning layer yet).
- Status remap for MedicationRequest is complete: `held→on-hold`, `discontinued→stopped`, unknown→`unknown`. Matches the FHIR R4 value set.
- `requester` priority `ordering_provider_id ?? prescribed_by` correctly handles the legacy column split.
- RxNorm (`http://www.nlm.nih.gov/research/umls/rxnorm`) and SNOMED route (`http://snomed.info/sct`) URLs are canonical.
- Practitioner emitted only for `physician | specialist | nurse` via `isClinicalRole()` — correctly filters `admin` / `family_caregiver` / `patient`.
- Name parser handles `"Sarah Jones"`, `"Sarah M. Jones"`, `"Sarah Jones, MD"` (credentials preserved in `text`), and single-token names.
- `password_hash` and `mfa_secret` are not surfaced — PHI boundary respected.
- Zero Attribution verified on commits and PR body.
- CI: Build, Test, Typecheck all pass.

### Findings: 0 critical, 4 suggestions, 1 nitpick

### Follow-up issues created
- **#942** — `urn:carebridge:users` identifier system should be a canonical URL (e.g. `https://carebridge.dev/fhir/sid/user-id`) for external-system interop.
- **#944** — Name parser mishandles Hispanic two-part surnames (`García López`), particle prefixes (`de Klerk`), multi-token hyphen-free family names. Proper fix is structured name columns on `users`.
- **#945** — **Hidden test-mock fragility.** `export-bundle-audit.test.ts`, `export-headers.test.ts`, `export-purge-warning.test.ts` were updated with `encounters` / `procedures` table mocks but NOT with `users` or `inArray`. Tests pass today only because the provider-ids Set stays empty. Any future fixture with `provider_id` populated will throw `inArray is not a function`. Defensive mock recommended.
- **#946** — `doseQuantity.code` reuses `dose_unit` verbatim under `system: http://unitsofmeasure.org`, which violates UCUM when `dose_unit` is `"tablet"`, `"puff"`, `"drop"`, `"IU"`, etc. Needs UCUM allowlist + fallback to text-only unit.
- **#947** — US Core Practitioner conformance (NUCC specialty coding, NPI identifier, `meta.profile` stamp). Defer to compliance pass.

### Nitpick
- `medication-request.ts` line 789: `"proposed"` maps to `"draft"` in status. `proposed` is not in the CareBridge medication status enum today — the case is dead code. Harmless but worth trimming or documenting as forward-compat.

### Rebase required
**YES** — PR #922 has not merged yet; this branch (`feat/issue-388-fhir-practitioner-medicationrequest`) stacks on it. Once #922 lands, rebase #923 onto `main`. Expected clean rebase (only router.ts import order and generators/index.ts exports are shared surfaces).

### Key concerns
- **Test mock fragility is the most important follow-up** — it's a latent trap, not a live bug. Consider rolling the fix into this PR or a fast-follow, since it touches the same test files you just modified.
- Everything else is MVP-acceptable and tracked.

File: services/fhir-gateway/src/router.ts [hunk 1]
```diff
@@ -398,6 +410,62 @@ export const fhirGatewayRouter = t.router({
...
+          });
+        }
+
+        // Epic-compatible MedicationRequest resources for active/ordered
+        // medications (#388). MedicationStatement above remains for the
+        // recorded/historical view; both formats are emitted so bundles
+        // round-trip cleanly in either direction.
+        for (const med of patientMedications) {
+          entry.push({
+            fullUrl: `urn:uuid:request-${med.id}`,
```
[the remark this answers is not in the document]
Comment (blamechris) [reply services/fhir-gateway/src/router.ts]: **FIX**

Fixed in `af7a219`. MedicationRequest bundle-entry fullUrl now uses a real `crypto.randomUUID()` under `urn:uuid:` instead of the invalid `urn:uuid:request-<id>` form. `resource.id` still carries `med.id` so the Patient subject reference and any cross-resource references stay stable.

```diff
  for (const med of patientMedications) {
    entry.push({
-     fullUrl: `urn:uuid:request-${med.id}`,
+     // urn:uuid: values must be real UUIDs per RFC 4122 …
+     fullUrl: `urn:uuid:${crypto.randomUUID()}`,
      resource: toFhirMedicationRequest(med, patientId),
    });
  }
```

File: services/fhir-gateway/src/__tests__/export-bundle-audit.test.ts [hunk 2]
```diff
@@ -69,6 +71,8 @@ vi.mock("@carebridge/db-schema", () => ({
   medications: medicationsTable,
   diagnoses: diagnosesTable,
   allergies: allergiesTable,
+  encounters: encountersTable,
+  procedures: proceduresTable,
 }));
```
[the remark this answers is not in the document]
Comment (blamechris) [reply services/fhir-gateway/src/__tests__/export-bundle-audit.test.ts:78]: **FIX**

Fixed in `af7a219`. `export-bundle-audit.test.ts` mocks extended with:
- `users: usersTable` on the `@carebridge/db-schema` mock
- `inArray: (_col, _values) => ({ __inArray: true })` on the `drizzle-orm` mock
- `toFhirPractitioner`, `toFhirMedicationRequest`, and `isClinicalRole` on the generators mock

File: services/fhir-gateway/src/__tests__/export-headers.test.ts [hunk 3]
```diff
@@ -54,6 +56,8 @@ vi.mock("@carebridge/db-schema", () => ({
   medications: medicationsTable,
   diagnoses: diagnosesTable,
   allergies: allergiesTable,
+  encounters: encountersTable,
+  procedures: proceduresTable,
 }));
```
[the remark this answers is not in the document]
Comment (blamechris) [reply services/fhir-gateway/src/__tests__/export-headers.test.ts:63]: **FIX**

Fixed in `af7a219`. Same mock extensions applied to `export-headers.test.ts`.

File: services/fhir-gateway/src/__tests__/export-purge-warning.test.ts [hunk 4]
```diff
@@ -57,6 +59,8 @@ vi.mock("@carebridge/db-schema", () => ({
   medications: medicationsTable,
   diagnoses: diagnosesTable,
   allergies: allergiesTable,
+  encounters: encountersTable,
+  procedures: proceduresTable,
 }));
```
[the remark this answers is not in the document]
Comment (blamechris) [reply services/fhir-gateway/src/__tests__/export-purge-warning.test.ts:66]: **FIX**

Fixed in `af7a219`. Same mock extensions applied to `export-purge-warning.test.ts`.

**Comment by blamechris** [comment]
## Review Comments Addressed

| # | Comment | Outcome | Reference |
|---|---------|---------|-----------|
| 1 | `urn:uuid:request-<id>` is an invalid urn:uuid value | FIX | `af7a219` |
| 2 | export-bundle-audit.test.ts mocks missing users/inArray/Practitioner/MedicationRequest | FIX | `af7a219` |
| 3 | export-headers.test.ts missing same | FIX | `af7a219` |
| 4 | export-purge-warning.test.ts missing same | FIX | `af7a219` |

**Total:** 4 comments addressed (plus 1 additional bug caught during this pass)
- Fixed: 4 review comments + 1 self-caught bug (case-sensitive MedicationRequest status mapper hid non-lowercase statuses — regression test added)
- False positives: 0
- Follow-up issues: 0

**Self-caught while processing the review:** `mapRequestStatus` didn't lowercase its input, so a `medications.status` row carrying anything other than lowercase (e.g. `"Active"` from a FHIR import) silently mapped to `"unknown"`, hiding the prescription in the export. Sibling generators (medication-statement, procedure, encounter) all lowercase before switching. Aligned and added 3 regression cases.

Earlier agent-review follow-ups remain open:
- #942 (`urn:carebridge:users` → URL-form identifier system)
- #944 (Hispanic / particle / multi-token surname parsing)
- #946 (UCUM dose_unit conformance audit)
- #947 (US Core Practitioner profile)
